pkg/gameservers/health.go

```diff
@@ -198,8 +198,28 @@ func (hc *HealthController) syncGameServer(ctx context.Context, key string) erro
 		return nil
 	}
 
-	if skip, err := hc.skipUnhealthy(gs); err != nil || skip {
-		return err
+	// retrieve the pod for the gameserver
+	pod, err := hc.podLister.Pods(gs.ObjectMeta.Namespace).Get(gs.ObjectMeta.Name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
I think this would be clearer (not withstanding the double negative) as:

```
if err != nil {
	if !k8serrors.IsNotFound(err) {
		// If the pod exists but there is an error, go back into the queue.
		return errors.Wrapf(err, "error retrieving Pod %s for GameServer to check status", gs.ObjectMeta.Name)
	}
	hc.baseLogger.WithField("gs", gs.ObjectMeta.Name).Debug("Could not find Pod")
}
```

which makes it clearer that one case _doesn't_ return and instead continues control flow down the function.